tools/cmd/seidb/operations/state_size.go

```diff
@@ -50,109 +53,246 @@ func executeStateSize(cmd *cobra.Command, _ []string) {
 		panic(err)
 	}
 	defer db.Close()
-	fmt.Printf("Finished opening db, calculating state size for module: %s\n", module)
-	err = PrintStateSize(module, db)
-	if err != nil {
-		panic(err)
+
+	// Get the actual height of the opened database
+	actualHeight := db.Version()
+	fmt.Printf("Finished opening db at height %d (requested: %d), calculating state size for module: %s\n", actualHeight, height, module)
+
+	// First, collect all the data by scanning the trees
+	moduleResults := collectAllModuleData(module, db)
+
+	// Then process the results based on the flag
+	if exportDynamoDB {
+		fmt.Printf("Exporting results to DynamoDB table: %s\n", dynamoDBTable)
+		err = exportResultsToDynamoDB(moduleResults, actualHeight, dynamoDBTable, awsRegion)
+		if err != nil {
+			panic(err)
+		}
+		fmt.Println("Successfully exported to DynamoDB!")
+	} else {
+		printResultsToConsole(moduleResults)
 	}
 }
 
-// PrintStateSize print the raw keys and values for given module at given height for memIAVL tree
-func PrintStateSize(module string, db *memiavl.DB) error {
+// collectModuleStats collects all the statistics for a module
+func collectModuleStats(tree *memiavl.Tree, moduleName string) *ModuleResult {
+	result := &ModuleResult{
+		ModuleName:        moduleName,
+		KeySizeByPrefix:   make(map[string]int64),
+		ValueSizeByPrefix: make(map[string]int64),
+		TotalSizeByPrefix: make(map[string]int64),
+		NumKeysByPrefix:   make(map[string]int64),
+		ContractSizes:     make(map[string]*utils.ContractSizeEntry),
```

Comment:
Questions:
* How big can these maps get in terms of memory footprint?
* do we need the exact prefix values or simply their distribution? If latter, hash them to cap the maximum memory footprint?

Reply:
These maps are pretty small, there are usually less than 10 different key prefix. Each key prefix is like 2 bytes. We need the exact prefix values, since are pretty small, shouldn't be a big concern, but yeah, this is a good point to think about